bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1089 @@
+  phase of the transition to the new gossip protocol could be very beneficial 
+  since nodes can continue to broadcast their legacy `node_announcement` and 
+  thus more effectively advertise their support for `option_taproot_gossip`. 
+- Nodes are encouraged to actively connect to other nodes that advertise the 
+  `option_taproot_gossip` feature bit as this is the only way in which they 
+  will learn about taproot channel announcements and updates.
+- Nodes should not use the new `node_announcement_2` message until they have
+  no more announced legacy channels.
+
+### Alternative Approaches
+
+An alternative approach would be to add an optional TLV to the legacy
+`node_annoucement` that includes the serialised `node_announcement_2`. The nice
+thing about this is that it allows the new `node_announcement_2` message to
+propagate quickly through the network before the upgrade is complete. The
+downside of this approach is that it would more than double the size of the
+legacy `node_announcement` and most of the information would be duplicated. It
+also makes gossip queries tricky since `node_announcement_2` uses a block-height
+based timestamp instead of the unix timestamp used by the legacy message.
+There also does not seem to be a big benefit in spreading the 
+`node_announcement_2` message quickly since the speed of propagation of the 
+`channel_announcement_2` and `channel_update_2` messages will still depend on a
+network wide upgrade.
+
+## Specification
+
+### Feature Bits
+
+Unlike the original set of gossip messages, it needs to be explicitly advertised
+that nodes are aware of the new gossip messages defined in this document at
+least until the whole network has upgraded to only using taproot gossip. We thus
+define a new feature bit, `option_taproot_gossip` to be included in the `init`
+message as well as the _old_ `node_announcement` message. Note that for all
+other feature bits with the `N` and `C` contexts, it can be assumed that those
+contexts will switch to the new `node_announcement_2` and
+`channel_announcement_2` messages for all feature bits _except_ for
+`option_taproot_gossip` which only makes sense in the context of the old
+`node_announcement` message and not the new `node_announcement_v2` message.
+The `option_taproot` feature bit can also be implied when using the new taproot 
+gossip messages. 
+
+| Bits  | Name                    | Description                                       | Context | Dependencies     |
+|-------|-------------------------|---------------------------------------------------|---------|------------------|
+| 32/33 | `option_taproot_gossip` | Nodes that understand the taproot gossip messages | IN      | `option_taproot` | 
+
+A node can be assumed to understand the new gossip v2 messages if:
+
+- They advertise `option_taproot_gossip` in the `init` or 
+  legacy `node_announcement` messages
+  OR
+- They have broadcast one of the new gossip messages defined in this document.
```

Comment:
I think it's probably a good idea to require consistency here. I don't love the idea of a node broadcasting Gossip v2 messages without having advertised `option_taproot_gossip` at all. I'm open to being challenged here. I think it's fair to say that if they broadcast it, they do understand it, as you note. However, I think things get significantly more complicated on the implementation side if we have to infer the node's ability to understand from either of these somewhat independent informational sources.

Reply:
agreed - I've removed this 👍